tutorials/data-science-preprocessing/index.md

```diff
@@ -0,0 +1,437 @@
+Cloud Dataflow which can handle orders of magnitude more data at a time.
+
+Because this sample uses a `BigQuerySink` for its output, you must first perform
+some setup:
+
+* First, make sure you're using the correct project:
+
+      $ gcloud config set project your-project-id
+      Updated property [core/project]
+      $
+
+* Now, create a BigQuery dataset using the `bq` command provided by the Google
+  Cloud SDK:
+
+      $ bq mk meteor_dataset
+      Dataset 'your-project-id:meteor_dataset' successfully created
+      $
+
+* To ensure that this example's dependencies don't conflict with any existing
+  libraries on your system, create a virtualenv and install the script
+  dependencies:
+
+      $ virtualenv venv
+      ...
+      $ source venv/bin/activate
+      (venv) $ pip install -r requirements.txt
+      ...
+      (venv) $
+
+With the BigQuery dataset created and your requirements installed, you are ready
+to run your pipeline.
+
+### Running locally
+
+    python clean.py meteors.json your-project-id:meteor_dataset.cleansed
+
+### Running on Cloud Dataflow
+
+To run the script on Cloud Dataflow, you'll have to specify your project, and a
+bucket in Google Cloud Storage:
+
+    PROJECT=your-project-id
+    # The $PROJECT.appspot.com bucket is created automatically with your project
+    BUCKET=gs://$PROJECT.appspot.com
+
+Now upload the json data file to Cloud Storage using the `gsutil` tool included
+in the Google Cloud SDK:
+
+    gsutil cp meteors.json gs://$BUCKET/
+
+Then run:
+
+    python clean.py gs://$BUCKET/meteors.json $PROJECT:meteor_dataset.cleansed \
+        --project $PROJECT \
+        --job_name $PROJECT-meteors \
+        --runner BlockingDataflowPipelineRunner \
+        --staging_location $BUCKET/staging \
+        --temp_location $BUCKET/temp \
+        --output $BUCKET/output
+
+You can check the status of your job on the [Dataflow
+dashboard](https://console.cloud.google.com/dataflow?project=_).
+
+## Sample query
+
+Now that the data has been ingested into BigQuery, it's available to explore.
+We'll go into more detail about this in the next section on [data
+exploration](exploration.md), but to give you an idea, here is a sample query we
+could do to find the number of meteor landings per year:
+
+     $ bq query "select year, count(*) from \
+         [$PROJECT:meteor_dataset.cleansed] \
+         group by year \
+         order by year desc \
+         limit 10"
+     Waiting on bqjob_r2a3ebeea2f95498c_00000158939ae926_1 ... (0s) Current
+     status: DONE
+     +------+------+
+     | year | f0_  |
+     +------+------+
+     | 2013 |    1 |
+     | 2012 |   15 |
+     | 2011 |  391 |
+     | 2010 |  420 |
+     | 2009 |  323 |
+     | 2008 |  261 |
+     | 2007 |  236 |
+     | 2006 | 1223 |
+     | 2005 |  246 |
+     | 2004 |  264 |
+     +------+------+
+
+## API Documentation
+
+We've only touched on a couple of the capabilities of Apache Beam and Cloud
```

Comment:
Again, in a perfect world, you would keep this in the same second person voice as the rest of the doc.